src/main/java/fi/hsl/transitdata/tripupdate/models/PubtransData.java

```diff
@@ -0,0 +1,56 @@
+package fi.hsl.transitdata.tripupdate.models;
+
+import fi.hsl.common.transitdata.proto.InternalMessages;
+import fi.hsl.common.transitdata.proto.PubtransTableProtos;
+import fi.hsl.transitdata.tripupdate.processing.BaseProcessor;
+
+/**
+ * Intermediate data format which will be refactored out soon, we'll just read StopEstimates directly.
+ */
+public class PubtransData {
+
+    public PubtransTableProtos.Common common;
+    public PubtransTableProtos.DOITripInfo tripInfo;
+    public BaseProcessor.EventType eventType;
+
+    public PubtransData(BaseProcessor.EventType eventType, PubtransTableProtos.Common common, PubtransTableProtos.DOITripInfo tripInfo) {
+        this.tripInfo = tripInfo;
+        this.common = common;
+        this.eventType = eventType;
+    }
+
+    /**
+     * StopEstimate is the data-model that we're going to use in the future so let's refactor this codebase already to use it internally.
+     * Then it's easy to swap to new input schema.
+     *
+     * Throws RuntimeException if data is not valid.
+     */
+    public InternalMessages.StopEstimate toStopEstimate() throws Exception {
+        InternalMessages.StopEstimate.Builder builder = InternalMessages.StopEstimate.newBuilder();
+        builder.setSchemaVersion(builder.getSchemaVersion());
+
+        InternalMessages.TripInfo.Builder tripBuilder = InternalMessages.TripInfo.newBuilder();
+        tripBuilder.setTripId(Long.toString(tripInfo.getDvjId()));
+        tripBuilder.setOperatingDay(tripInfo.getOperatingDay());
+        tripBuilder.setRouteId(tripInfo.getRouteId());
+        tripBuilder.setDirectionId(tripInfo.getDirectionId());//Jore format
```

Comment:
Does it make sense to call `direction_id` `jore_direction_id` in the `internal-messages.proto`?

Reply:
this was discussed but there wasn't a clear call which was better... jore is quite context-specific term also.. decided that we leave it to comments and convention. time will tell if that was a good decision.